x/ccv/provider/keeper/key_assignment_test.go

```diff
@@ -0,0 +1,494 @@
+package keeper_test
+
+import (
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	cryptotestutil "github.com/cosmos/interchain-security/testutil/crypto"
+	testkeeper "github.com/cosmos/interchain-security/testutil/keeper"
+	"github.com/stretchr/testify/require"
+	abci "github.com/tendermint/tendermint/abci/types"
+	tmprotocrypto "github.com/tendermint/tendermint/proto/tendermint/crypto"
+)
+
+type testAssignment struct {
+	chainID        string
+	providerAddr   sdk.ConsAddress
+	consumerAddr   sdk.ConsAddress
+	consumerPubKey tmprotocrypto.PublicKey
+	valsetUpdate   abci.ValidatorUpdate
+	vscID          uint64
+}
+
+func TestValidatorConsumerPubKeyCRUD(t *testing.T) {
+	chainID := "consumer"
+	providerAddr := sdk.ConsAddress([]byte("providerAddr"))
+	consumerKey := cryptotestutil.NewCryptoIdentityFromIntSeed(1).TMProtoCryptoPublicKey()
+
+	keeper, ctx, ctrl, _ := testkeeper.GetProviderKeeperAndCtx(t, testkeeper.NewInMemKeeperParams(t))
+	defer ctrl.Finish()
+
+	keeper.SetValidatorConsumerPubKey(ctx, chainID, providerAddr, consumerKey)
+
+	consumerPubKey, found := keeper.GetValidatorConsumerPubKey(ctx, chainID, providerAddr)
+	require.True(t, found, "consumer pubkey not found")
+	require.NotEmpty(t, consumerPubKey, "consumer pubkey is empty")
+	require.Equal(t, consumerPubKey, consumerKey)
+
+	keeper.DeleteValidatorConsumerPubKey(ctx, chainID, providerAddr)
+	consumerPubKey, found = keeper.GetValidatorConsumerPubKey(ctx, chainID, providerAddr)
+	require.False(t, found, "consumer pubkey was found")
+	require.Empty(t, consumerPubKey, "consumer pubkey was returned")
+	require.NotEqual(t, consumerPubKey, consumerKey)
+}
+
+func TestIterateValidatorConsumerPubKeys(t *testing.T) {
+
+	chainID := "consumer"
+	testAssignments := []testAssignment{
+		{
+			providerAddr:   sdk.ConsAddress([]byte("validator-1")),
+			consumerPubKey: cryptotestutil.NewCryptoIdentityFromIntSeed(1).TMProtoCryptoPublicKey(),
+		},
+		{
+			providerAddr:   sdk.ConsAddress([]byte("validator-2")),
```

Comment:
ditto - and in several other places

Reply:
@MSalopek if these nits are 'wont-fix' could you please resolve the comment boxes? Thank you!